songpal/main.py

```diff
@@ -250,6 +263,35 @@ def print_settings(settings, depth=0):
                 click.echo("    - %s" % out)
 
 
+@cli.command()
+@click.argument("zone", required=False)
+@click.argument("activate", required=False, type=click.BOOL)
+@pass_dev
+@coro
+async def zone(dev: Device, zone, activate):
+    """Get and change outputs."""
+    zones = await dev.get_zones()
+    if zone:
+        try:
+            zone = next((x for x in zones if x.title == zone))
+            
+            action = "Activating" if activate else "Deactivating"
+
+            click.echo("%s %s" % (action, zone)) 
```

Comment:
Same here as above, make get_zones() to raise an error if no zones is available -> error handling comes automatically. And if zone name is defined, using the `get_zone(name)` should work, I suppose, making it necessary only to have the loop for printing out all the zones.

Reply:
I believe this is resolved